pkg/cloud/amazon/eks/eks.go

```diff
@@ -13,34 +17,30 @@ import (
 	"github.com/pkg/errors"
 )
 
-// EksClusterOptions contains some functions to interact with EKS
-type EksClusterOptions struct {
+// eksAPIHandler contains some functions to interact with and serves as an abstraction of the EKS API
+type eksAPIHandler struct {
 	eks eksiface.EKSAPI
 }
 
-// NewEKSClusterOptions will return an EksClusterOptions value and configure credentials
-func NewEKSClusterOptions(eksapi ...eksiface.EKSAPI) (*EksClusterOptions, error) {
+// NewEKSAPIHandler will return an eksAPIHandler with configured credentials
+func NewEKSAPIHandler(awsSession client.ConfigProvider, eksapi ...eksiface.EKSAPI) (*eksAPIHandler, error) {
 	if len(eksapi) == 1 {
-		return &EksClusterOptions{
+		return &eksAPIHandler{
 			eks: eksapi[0],
 		}, nil
 	}
-	session, err := NewAwsSession("", "")
-	if err != nil {
-		return nil, errors.Wrap(err, "there was a problem ensuring the initialization of the EKS API")
-	}
-	return &EksClusterOptions{
-		eks: eks.New(session),
+	return &eksAPIHandler{
+		eks: eks.New(awsSession),
 	}, nil
 }
 
 // EksClusterExists checks if EKS cluster with given name exists in given region.
-func (e *EksClusterOptions) EksClusterExists(clusterName string, profile string, region string) (bool, error) {
-	region, err := ResolveRegion(profile, region)
+func (e *eksAPIHandler) EksClusterExists(clusterName string, profile string, region string) (bool, error) {
```

Comment:
what's about the _profile_ parameter?

Reply:
That's the AWS credentials profile. If the user wants to provide a certain profile to interact with AWS through Jenkins X we should use it.

Reply:
Sure, what I meant is that we should add this to the docs.